src/test/java/seedu/address/model/AddressBookTest.java

```diff
@@ -76,6 +77,39 @@ public void hasTag_tagWithSameIdentityFieldsInAddressBook_returnsTrue() {
         assertTrue(addressBook.hasTag(editedCS2103));
     }
 
+
+    @Test
+    public void setTag_tagInAddressBook_returnsTrue() {
+        AddressBook dummyAddressBook = addressBook;
+        dummyAddressBook.addTag(CS2103);
+        dummyAddressBook.setTag(CS2103, CS2101);
+        Tag expectedTag = new TagBuilder(CS2101).build();
+        assertTrue(dummyAddressBook.hasTag(expectedTag));
+    }
+
+    @Test
+    public void setTag_nullTag_returnsNullPointerException() {
+        AddressBook dummyAddressBook = addressBook;
+        dummyAddressBook.addTag(CS2103);
+        assertThrows(NullPointerException.class, () -> dummyAddressBook.setTag(CS2103, null));
+    }
+
+    @Test
+    public void removeTag_tagInAddressBook_returnsFalse() {
```

Comment:
Should the method name be `removeTag_tagInAddressBook_success`? Since the result of the method being tested is not actually returning false.

Reply:
Sure!